python/pyspark/sql/session.py

```diff
@@ -1024,7 +1024,14 @@ def prepare(obj: Any) -> Any:
         if isinstance(data, RDD):
             rdd, struct = self._createFromRDD(data.map(prepare), schema, samplingRatio)
         else:
-            rdd, struct = self._createFromLocal(map(prepare, data), schema)
+            if isinstance(data, list):
+                # Wrap each element with a tuple if there is any scalar in the list
+                has_scalar = any(isinstance(x, str) or not hasattr(x, "__len__") for x in data)
```

Comment:
`hasattr(x, "__len__")` could be replaced to `isinstance(x, collections.abc.Sized)`